### DISCUSSION TITLE
Requesting the window to be closed?

### DISCUSSION THREAD

**Discussion by npwoods** [slint-ui/slint#5465] [Q&A] [answered]
When I close out a Slint window, the callback specified by `on_close_requested()` will be invoked.  How do I initiate this process manually from code, so I could have say a "quit" button or an "Exit" menu item?  `quit_event_loop()` bypasses all of this apparatus.

**Comment by ogoffart** [comment, marked as answer]
on_close_requested() is to be used when the close request comes from the window manager.
You can simulate this with a slint::platform::WindowEvent::CloseRequested event. But I wouldn't recommand to do that for the "Exit" menu item. For that, i'd recommend that the callback handler for Exit and the for on_close_requested call the same helper function.
